src/difference/difference_solution1/difference.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * This function creates an array of values that are in the first array,
+ * but not not in the second array.
+ *
+ * Repeated values are not duplicated in the return value,
+ * and the order of result values are determined by the first array.
+ *
+ * **Note:** This function returns a new array, and has no side-effects.
+ *
+ * @param {Array} [arrayToFindTheDifference=[]] - The array to inspect.
+ * @param {Array} [arrayToCompare=[]] - The values to exclude.
+ *
+ * @returns {Array} Returns the new array of filtered values.
+ *
+ * @example
+ *
+ * difference([2, 1], [2, 3]); //  [1]
+ *
+ * @example
+ *
+ * difference([1, 2, 1], [2, 3]); //  [1]
+ */
+export const arrayDifference = (
+  arrayToFindTheDifference = [],
+  arrayToCompare = [],
+) => {
+  // ensure that a new array is created
```

Comment:
To many comments, as a coder you only need comments only in the places that you may think is going to be tricky to debug for future developers, as coaches say to many comments give the impression that the code is hard to understand

Reply:
JS doc is the documentation and it will be used by intellicence to guide you while you write the code. So the more the better (up to a point). But I totally agree with the comments in the function body, but the check list 
" Comments are used to label important goals in your strategy
 Comments are not used to describe how the code works" 
 forces these kind of things in small methods. Any suggestions how these can be done in this method?